src/ziggurat/mapper.clj

```diff
@@ -23,18 +23,23 @@
                 end-time    (.toEpochMilli (Instant/now))]
             (metrics/report-time (str topic-entity-name ".handler-fn-execution-time") (- end-time start-time))
             (case return-code
-              :success (metrics/increment-count metric-namespace "success")
-              :retry (do (metrics/increment-count metric-namespace "retry")
-                         (producer/retry message topic-entity))
-              :skip (metrics/increment-count metric-namespace "skip")
-              :block 'TODO
+              :success (do (metrics/increment-count metric-namespace "success" topic-entity-name)
+                           (metrics/increment-count "message-processing" "success" topic-entity-name))
+              :retry   (do (metrics/increment-count metric-namespace "retry" topic-entity-name)
+                           (metrics/increment-count "message-processing" "retry" topic-entity-name)
+                           (producer/retry message topic-entity))
+              :skip    (do (metrics/increment-count metric-namespace "skip" topic-entity-name)
+                           (metrics/increment-count "message-processing" "skip" topic-entity-name))
+              :block   'TODO
               (do
                 (send-msg-to-channel channels message topic-entity return-code)
-                (metrics/increment-count metric-namespace "success"))))
+                (metrics/increment-count metric-namespace "success" topic-entity-name)
+                (metrics/increment-count "message-processing" "success" topic-entity-name))))
```

Comment:
`(metrics/increment-count "message-processing" "success" {:topic-entity-name "topic-entity-name"})`

Reply:
wdyt @theanirudhvyas

Reply:
done

Reply:
yes, makes sense, this could be a custom-tags map where users can also add their own custom tags to metrics.